### ISSUE TITLE
Add UML Diagrams

### ISSUE THREAD

**Issue by torbenkeller** [scottt2/design-patterns-in-dart#3]
I think especially for programmers who don't know exactly how each pattern works, it would be nice to have some UML Diagrams to make the concept more clear

**Comment by scottt2** [comment]
@torbenkeller I agree! I will add it to my TODO. Maybe start small and provide some examples so other folks can also contribute. I'll keep this open and update the progress here. Thanks for the idea!
